extensions/default/src/ViewerLayout/ResizablePanelsHook.tsx

```diff
@@ -89,103 +99,47 @@ const useResizablePanels = (
-    // of the dependency on the  `resizablePanelDefaultSize` state, this code
-    // is executed on the render following an update of the minimum percentage size
+    // updated which occurs only AFTER the render following a browser window resize.
+    // And by virtue of the dependency on the minimum size state variables, this code
+    // is executed on the render following an update of the minimum percentage sizes
     // for a panel.
     if (!resizableLeftPanelAPIRef.current.isCollapsed()) {
-      const leftSize =
-        ((leftPanelExpandedWidth + panelGroupDefinition.shared.expandedInsideBorderSize) /
-          panelGroupWidth) *
-        100;
+      const leftSize = getPercentageSize(
+        leftPanelExpandedWidth + panelGroupDefinition.shared.expandedInsideBorderSize
+      );
       resizableLeftPanelAPIRef.current.resize(leftSize);
     }
 
     if (!resizableRightPanelAPIRef.current.isCollapsed()) {
-      const rightSize =
-        ((rightPanelExpandedWidth + panelGroupDefinition.shared.expandedInsideBorderSize) /
-          panelGroupWidth) *
-        100;
+      const rightSize = getPercentageSize(
+        rightPanelExpandedWidth + panelGroupDefinition.shared.expandedInsideBorderSize
+      );
       resizableRightPanelAPIRef.current.resize(rightSize);
     }
 
     // This observer kicks in when the ViewportLayout resizable panel group
     // component is resized. This typically occurs when the browser window resizes.
     const observer = new ResizeObserver(() => {
-      const { width: panelGroupWidth } = resizablePanelGroupElemRef.current.getBoundingClientRect();
-      const minimumLeftSize =
-        (panelGroupDefinition.left.minimumExpandedOffsetWidth / panelGroupWidth) * 100;
-      const minimumRightSize =
-        (panelGroupDefinition.right.minimumExpandedOffsetWidth / panelGroupWidth) * 100;
-
-      // Set the new default and collapsed resizable panel sizes.
-      setLeftResizablePanelMinimumSize(Math.min(50, minimumLeftSize));
-      setRightResizablePanelMinimumSize(Math.min(50, minimumRightSize));
+      const minimumLeftSize = getPercentageSize(
+        panelGroupDefinition.left.minimumExpandedOffsetWidth
+      );
+      const minimumRightSize = getPercentageSize(
+        panelGroupDefinition.right.minimumExpandedOffsetWidth
+      );
+
+      // Set the new minimum and collapsed resizable panel sizes.
+      setLeftResizablePanelMinimumSize(minimumLeftSize);
+      setRightResizablePanelMinimumSize(minimumRightSize);
       setLeftResizePanelCollapsedSize(
-        (panelGroupDefinition.left.collapsedOffsetWidth / panelGroupWidth) * 100
+        getPercentageSize(panelGroupDefinition.left.collapsedOffsetWidth)
       );
       setRightResizePanelCollapsedSize(
-        (panelGroupDefinition.right.collapsedOffsetWidth / panelGroupWidth) * 100
+        getPercentageSize(panelGroupDefinition.right.collapsedOffsetWidth)
       );
-
```

Comment:
This chunk of code is from an earlier implementation and is no longer needed. It was an attempt to handle an extreme edge case when the viewport grid is zero width. However it is such an extreme edge case that actually is handled fairly well by the underlying library now that we respond better to window resizing.